jekyll/_cci2/test-config-policies.adoc

```diff
@@ -0,0 +1,613 @@
+   {
+     "enabled_rules": [
+-      "check_min_remote_docker_version",
++      "check_version",
+-      "check_version"
+     ],
+     "status": "PASS"
+   }
+FAIL    test_version_wrong_type    0.000s
+   {
+     "enabled_rules": [
+-      "check_min_remote_docker_version",
++      "check_version",
+-      "check_version"
+     ],
+     "hard_failures": [{"reason":"version must be a number","rule":"check_version"}],
+     "status": "HARD_FAIL"
+   }
+fail    config-policies    0.001s
+
+0/7 tests passed (0.001s)
+Error: unsuccessful run
+----
+
+Adding a new policy to the bundle added a new rule, which led to the failures. The decision in two ways:
+
+- A new rule was added to the `enabled_rules` field
+- A new `soft_failure` occurs because not all of the tests specify the configuration `version` as it is not needed for the Docker version policy.
+
+The following section introduces policy file structure best practices for managing your policies, to avoid this problem.
+
+[#manage-policy-test-file-structure]
+== Manage policy test file structure
+
+When the `circleci policy test` command is pointed at a folder, for example `./config-policies`, it will pick up every `*_test.yaml` file in that folder, and run those tests against the policy **rooted** at that folder.
+
+It is best-practice to use a file structure that allows you to write stable tests for individual policies, as well as tests for the full policy bundle, as follows:
+
+[source,shell]
+----
+├── config-policies/
+│   ├── policy_test.yaml
+│   ├── policy1/
+│   │   ├── policy1.rego
+│   │   ├── policy1_test.rego
+│   ├── policy2/
+│   │   ├── policy2.rego
+│   │   ├── policy2_test.rego
+----
+
+
+It is a good idea to have tests that run against the entire bundle that will be active in production, but we also want to be able to write stable tests against an individual policy. This is achieved by isolating each policy in its own subfolder with its tests. This way each subfolder will run with a sub-bundle and the tests defined within it.
+
+. Update the file structure:
++
+[source,shell]
+----
+├── config-policies/
+│   ├── docker/
+│   │   ├── docker.rego
+│   │   ├── docker_test.rego
+│   ├──version/
+│   │   ├── version.rego
+│   │   ├── version_test.rego
+----
+
+. Run all all tests including those in subfolders by appending `/...` to the test path:
++
+[source,shell]
+----
+circleci policy test ./config-policies/...
+----
++
+Expected output. Tests are passing again:
++
+[source,shell]
+----
+?     config-policies            no tests
+ok    config-policies/docker     0.000s
+ok    config-policies/version    0.000s
+
+7/7 tests passed (0.001s)
+----
+
+. To build more confidence, best practice is to create a top level test that will use the entire policy bundle, similar to an integration or end-to-end test.
+. Create a new test file: ./config-policies/policy_test.yaml`
+. Copy the following into your `policy_test.yaml` file:
++
+[source,yaml]
+----
+test_policy:
```

Comment:
@davidmdm Could you check what changes I need to make for this end to end test. Now I've based the examples on following on from the basic "write a policy" how to guide I think there are some small changes needed.